src/Io/Transaction.php

```diff
@@ -104,7 +106,7 @@ private function next(RequestInterface $request, Deferred $deferred)
         $that = $this;
         ++$deferred->numRequests;
 
-        $promise = $this->sender->send($request);
+        $promise = $this->sender->send($request, array('decodeContent' => $this->decodeContent));
```

Comment:
No major implementation concerns, but I feel this implementation should live in this class instead of the message factory, no?

Reply:
My rationale was that the body of the HTTP response needed to be transformed (decoded) before the Response object is created, which takes place inside the MessageFactory class.